src/authenticators/mod.rs

```diff
@@ -54,18 +57,26 @@ pub trait Authenticate {
 impl ApplicationName {
     /// Create a new ApplicationName from a String
     pub fn new(name: String) -> ApplicationName {
-        ApplicationName(name)
+        ApplicationName {
+            name,
+            is_admin: false,
+        }
     }
 
     /// Get a reference to the inner string
     pub fn get_name(&self) -> &str {
-        &self.0
+        &self.name
+    }
+
+    /// Flag current application as being an admin
+    fn set_admin(&mut self) {
```

Comment:
What about this method taking a boolean to set the field? I feel like `set_admin(true)` is nice to understand

Reply:
I think I'll change my approach here and just keep `ApplicationName` immutable, though that means I'll have to check whether a string is an admin name before calling the constructor.